torch/onnx/symbolic_opset9.py

```diff
@@ -1505,6 +1505,15 @@ def sort(g, self, dim, decending, out=None):
 
     return g.op("TopK", self, k_i=self.type().sizes()[dim], axis_i=dim, outputs=2)
 
+
+def numel(g, self):
+    shape = g.op("Shape", self)
+    dtype = shape.type().scalarType()
+    if dtype is None:  # dtype could be None for a script module
```

Comment:
`shape` is just created above, I think here `dtype` would always `None`. 

On the other hand, according to onnx spec, `Shape` operator always output tensor of type `Long`, why do we need to insert the `cast` node?

Reply:
I was seeing the ONNXRuntime error for kernel not implemented, and I thought this missing type is the issue. But that's correct, it should not be neede.
This PR needs to wait until https://github.com/microsoft/onnxruntime/pull/3507 is merged then.